diff --git a/numpy/linalg/setup.py b/numpy/linalg/setup.py
index 1fb7a3a..6e3c779 100644
--- a/numpy/linalg/setup.py
+++ b/numpy/linalg/setup.py
@@ -11,14 +11,7 @@ def configuration(parent_package='',top_path=None):
     # Configure lapack_lite
     lapack_info = get_info('lapack_opt',0) # and {}
     def get_lapack_lite_sources(ext, build_dir):
-        if not lapack_info:
-            print("### Warning:  Using unoptimized lapack ###")
-            return ext.depends[:-1]
-        else:
-            if sys.platform=='win32':
-                print("### Warning:  python_xerbla.c is disabled ###")
-                return ext.depends[:1]
-            return ext.depends[:2]
+        return ext.depends[:2]
 
     config.add_extension('lapack_lite',
                          sources = [get_lapack_lite_sources],
